presto-docs/src/main/sphinx/security/salesforce.rst

```diff
@@ -0,0 +1,80 @@
+============================
+Salesforce Authentication
+============================
+
+Presto can be configured to enable frontend password authentication over
+HTTPS for clients, such as the CLI, or the JDBC and ODBC drivers. The
+username and password (or password and :ref:`security tokens` concatenation)
+are validated by having the Presto coordinator perform a login to Salesforce.
+
+This allows you to enable users to authenticate to Presto via their Salesforce
+basic credentials.  This can also be used to secure the :ref:`Web UI <web-ui-authentication>`.
+
+.. note::
+
+    This is *not* a Salesforce connector, and does not allow users to query
+    Salesforce data. It's simply a means by which users can authenticate to Presto, similar
+    to :doc:`ldap` or :doc:`password-file`.
+
+Salesforce Authenticator Configuration
+--------------------------------------
+
+Enable Salesforce authentication by creating an
+``etc/password-authenticator.properties`` file on the coordinator:
+
+.. code-block:: properties
+
+    password-authenticator.name=salesforce
+    salesforce.allowed-organizations=allowed salesforce org id's, or the word all
+
+The following configuration properties are available:
+
+====================================   ==============================================
+Property                               Description
+====================================   ==============================================
+``salesforce.allowed-organizations``   Comma separated list of 18 character Salesforce.com
+                                       Organization IDs for a second, simple layer of security.
+                                       This option can be explicitly ignored using ``all``, which
+                                       bypasses any check of the authenticated user's
+                                       Salesforce.com Organization ID.
+
+``salesforce.cache-size``              Max number of cached authenticated users.
+                                       Defaults to ``4096``.
+
+``salesforce.cache-expire-duration``   Duration for an entry to remain in the cache after the
+                                       first write before it is removed.
+                                       Defaults to ``2m``.
+====================================   ==============================================
+
+Salesforce Concepts
+-------------------
+
+There are two Salesforce specific aspects to this authenticator.  They are the use of the
+Salesforce security token, and configuration of one or more Salesforce.com Organization IDs.
+
+.. _security tokens:
```

Comment:
This creates a global reference which is problematic since "security tokens" is rather generic. Instead, we can link directly to this section since it's inside the same document, so no need for a reference at all. I gave the syntax above.

Reply:
Removing.